components/org.wso2.carbon.crypto.provider/src/main/java/org/wso2/carbon/crypto/provider/KeyStoreBasedInternalCryptoProvider.java

```diff
@@ -168,7 +168,15 @@ public KeyStoreBasedInternalCryptoProvider(KeyStore keyStore, String keyAlias, S
     public byte[] encrypt(byte[] cleartext, String algorithm, String javaSecurityAPIProvider,
                           boolean returnSelfContainedCipherText) throws CryptoException {
 
-        byte[] encryptedKey = encrypt(cleartext, algorithm, javaSecurityAPIProvider);
+        byte[] encryptedKey;
+        if (StringUtils.isNotBlank(algorithm) && cleartext.length == 0) {
```

Comment:
Do we need to handle ** cleartext** being null?

Reply:
This null is handled from kernel level [1]. An exception is thrown if the value is null, before invoking this method.


[1]https://github.com/wso2/carbon-kernel/blob/1b335215cb26b4280220cd498727016548bc9a86/core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java#L123

Reply:
Still , `cleartext.length` will give a NullPointer if `cleartext` is null